bundles/org.openhab.binding.roborock/src/main/java/org/openhab/binding/roborock/internal/RoborockAccountHandler.java

```diff
@@ -0,0 +1,489 @@
+        } else {
+            logger.warn("Initialized child handler is not a RoborockVacuumHandler: {}",
+                    childHandler.getClass().getName());
+        }
+    }
+
+    @Override
+    public void childHandlerDisposed(ThingHandler childHandler, Thing childThing) {
+        logger.debug("Child released from gateway: {}  {} -> {} {}", childThing.getUID(), childThing.getLabel(),
+                getThing().getUID(), getThing().getLabel());
+        childDevices.remove(childThing.getUID().getId());
+    }
+
+    private void initAPI() {
+        if (baseUri.isEmpty()) {
+            try {
+                baseUri = webTargets.getUrlByEmail(config.email);
+            } catch (RoborockException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Error " + e.getMessage());
+            }
+        }
+        String sessionStoreToken = sessionStorage.get("token");
+        String sessionStoreRriot = sessionStorage.get("rriot");
+
+        if (!(sessionStoreToken == null) && !(sessionStoreRriot == null)) {
+            logger.debug("Retrieved token and rriot values from sessionStorage");
+            token = sessionStoreToken;
+            @Nullable
+            Rriot rriotTemp = gson.fromJson(sessionStoreRriot, Rriot.class);
+            if (rriotTemp != null) {
+                rriot = rriotTemp;
+            }
+        } else {
+            logger.debug("No available token or rriot values from sessionStorage, logging in");
+            try {
+                String response = webTargets.doLogin(baseUri, config.email, config.password);
+                int code = 0;
+                if (!response.isEmpty() && JsonParser.parseString(response).getAsJsonObject().has("code")) {
+                    code = JsonParser.parseString(response).getAsJsonObject().get("code").getAsInt();
+                }
+                if (code == 200) {
+                    logger.debug("Successful login, parsing parameters");
+                    Login loginResponse = gson.fromJson(response, Login.class);
+                    sessionStorage.put("token", loginResponse.data.token);
+                    sessionStorage.put("rriot", gson.toJson(loginResponse.data.rriot));
+                    token = loginResponse.data.token;
+                    rriot = loginResponse.data.rriot;
+                    updateStatus(ThingStatus.ONLINE);
+                } else {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Unsuccessful login: "
+                            + JsonParser.parseString(response).getAsJsonObject().get("msg").getAsString());
+                    return;
+                }
+            } catch (RoborockException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Error " + e.getMessage());
+            }
+        }
+        updateStatus(ThingStatus.ONLINE);
```

Comment:
In the above the thing status is already set multiple times and this overwrites that state. Maybe make use of early returns or otherwise

Reply:
Yep good call - fixed.